pkg/sql/sem/tree/eval.go

```diff
@@ -877,6 +877,35 @@ var BinOps = map[BinaryOperator]binOpOverload{
 				return &DInterval{Duration: left.(*DInterval).Duration.MulFloat(r)}, nil
 			},
 		},
+		BinOp{
+			LeftType:   types.Float,
+			RightType:  types.Interval,
+			ReturnType: types.Interval,
+			fn: func(_ *EvalContext, left Datum, right Datum) (Datum, error) {
+				l := float64(*left.(*DFloat))
+				return &DInterval{Duration: right.(*DInterval).Duration.MulFloat(l)}, nil
+			},
+		},
+		BinOp{
+			LeftType:   types.Decimal,
```

Comment:
I'm surprised that the decimal variants are necessary for `select 2.0 * interval '1 second'`. The untyped constant should be recognized as a float in that case.

cc @knz

Reply:
Also I think the issue in this particular case is that the typing is ambiguous otherwise.